lib/cockpit-components-listing.jsx

```diff
@@ -17,257 +17,267 @@
-            };
-        },
-        render: function() {
-            var bodyClasses = ["listing", "listing-ct"];
-            if (this.props.fullWidth)
-                bodyClasses.push("listing-ct-wide");
-            var headerClasses;
-            var headerRow = <tr/>;
-            if (!this.props.children || this.props.children.length === 0) {
-                headerClasses = "listing-ct-empty";
-                headerRow = <tr><td>{this.props.emptyCaption}</td></tr>;
+            var listingDetail;
+            if ('listingDetail' in this.props) {
+                listingDetail = (
+                    <span className="listing-ct-caption">
+                        {this.props.listingDetail}
+                    </span>
+                );
             }
+
             return (
-                <table className={ bodyClasses.join(" ") }>
-                    <caption className="cockpit-caption">{this.props.title}</caption>
-                    <thead className={headerClasses}>
-                        {headerRow}
-                    </thead>
-                    {this.props.children}
-                </table>
+                <tbody className="open">
+                    {listingItem}
+                    <tr className="listing-ct-panel">
+                        <td colSpan={ headerEntries.length + (expandToggle?1:0) }>
+                            <div className="listing-ct-head">
+                                <div className="listing-ct-actions">
+                                    {listingDetail}
+                                    {this.props.listingActions}
+                                </div>
+                                <ul className="nav nav-tabs nav-tabs-pf">
+                                    {links}
+                                </ul>
+                            </div>
+                            {tabs}
+                        </td>
+                    </tr>
+                </tbody>
             );
-        },
-    });
+        } else {
+            return (
+                <tbody>
+                    {listingItem}
+                    <tr className="listing-ct-panel"/>
+                </tbody>
+            );
+        }
+    }
+});
 
-    return {
-        ListingRow: ListingRow,
-        Listing: Listing,
-    };
+/* Implements a PatternFly 'List View' pattern
+ * https://www.patternfly.org/list-view/
+ * Properties:
+ * - title
+ * - fullWidth optional: set width to 100% of parent, defaults to true
+ * - emptyCaption header caption to show if list is empty, defaults to "No entries"
+ * - columnTitles: array of column titles, as strings
+ * - actions: additional listing-wide actions (displayed next to the list's title)
+ */
+var Listing = React.createClass({
+    propTypes: {
+        title: React.PropTypes.string.isRequired,
+        fullWidth: React.PropTypes.bool,
+        emptyCaption: React.PropTypes.string.isRequired,
+        columnTitles: React.PropTypes.arrayOf(React.PropTypes.string)
```

Comment:
Added. Thanks. These are kind of pointless, but I like them for documentation purposes. I'd be ok with removing all of them, though, if nobody agrees.